### PR TITLE
runtime/block: send bools to the done port, accept bools

### LINKED ISSUE #4775: `block::system_handler` expects PMT long, gets PMT bool
Investigation #4770  I found something confusing:

`block.cc`:
```c++
void block::system_handler(pmt::pmt_t msg)
{
…
        d_finished = pmt::to_long(pmt::cdr(msg));
…}
```

notice the `to_long` in there: that fails on PMTs that `is_bool()`

in `notify_msg_neighbors`, however:

```c++
blk->post(d_system_port, pmt::cons(d_pmt_done, pmt::mp(true)));
```

Is it possible we currently can't shut down message-based flow graphs or am I misreading this?

### PR DESCRIPTION
**Pull request by marcusmueller**
but continue accepting numbers for compatibility

fixes #4775

### CODE REVIEW

**Review by jacobagilbert — COMMENTED** [review]
This needs to be fixed, but is there any reason to handle something other than a PMT bool? I would consider that [after this change] a non-bool cdr() sent to this port to be a bug.

**Comment by marcusmueller** [comment]
@jacobagilbert  thing is that sending a pmt::cons(pmt::mp("done"), pmt::mp(false)) to the system port is a way to "unblock" a block that's been blocked by a previous iteration where it produced zero items. 

Hence, that's a pattern that SDR driver OOTs might have used. So, it's API. Let me show what I've got in mind with another commit here.

**Comment by jacobagilbert** [comment]
I get why one would want to send `pmt::PMT_F` but i think what you are saying is that we need to accept `pmt::mp(0)` for now for compatibility also.

Sorry for 20 questions...but what is the reason to switch to accepting dicts() on this port?

**Comment by marcusmueller** [comment]
> Sorry for 20 questions...

Um, never be sorry for asking question!

> but what is the reason to switch to accepting dicts() on this port?

Consistency – other message control ports accept dicts. (the question maybe is whether we need a "system" port that accept "command dicts or pairs" which really can only contain one key)

File: gnuradio-runtime/lib/block.cc
Comment (jacobagilbert) [inline gnuradio-runtime/lib/block.cc:627]: maybe simplify:
```c++
if (pmt::is_bool(value)) {
  d_finished = pmt::to_bool(value);
} else {
  std::cout << "WARNING: bad message op in dict on system port!\n";
  return;
}
```

File: gnuradio-runtime/lib/block.cc
Comment (jacobagilbert) [inline gnuradio-runtime/lib/block.cc:636]: is `d_pmt_done` the only thing this port can handle?

File: gnuradio-runtime/lib/block.cc
Comment (jacobagilbert) [inline gnuradio-runtime/lib/block.cc:678]: Probably should change to
```c++
blk->post(d_system_port, pmt::dict_add(pmt.make_dict(), d_pmt_done, pmt::PMT_T));
```
if thats the recommendation

**Review by jacobagilbert — CHANGES_REQUESTED** [review]

File: gnuradio-runtime/lib/block.cc
Comment (marcusmueller) [reply gnuradio-runtime/lib/block.cc:636]: yep, at this point that's true – but I'm sure the intention was to control more through that port. Honestly, don't know whether that's wise.

File: gnuradio-runtime/lib/block.cc
Comment (marcusmueller) [reply gnuradio-runtime/lib/block.cc:627]: I like simple.

**Comment by mormj** [comment]
This seems to stall out on the message passing QA tests

**Comment by mormj** [comment]
@marcusmueller - I like the added flexibility that this gives the system port, but going to close this one for now since it breaks the message passing qa tests.

**Pull request closed without merging**
